tests/sundials/arkode_repeated_solve.cc

```diff
@@ -0,0 +1,136 @@
+//-----------------------------------------------------------
+//
+//    Copyright (C) 2021 by the deal.II authors
+//
+//    This file is part of the deal.II library.
+//
+//    The deal.II library is free software; you can use it, redistribute
+//    it, and/or modify it under the terms of the GNU Lesser General
+//    Public License as published by the Free Software Foundation; either
+//    version 2.1 of the License, or (at your option) any later version.
+//    The full text of the license can be found in the file LICENSE.md at
+//    the top level directory of deal.II.
+//
+//-----------------------------------------------------------
+
+#include <deal.II/base/parameter_handler.h>
+
+#include <deal.II/lac/full_matrix.h>
+#include <deal.II/lac/vector.h>
+
+#include <deal.II/sundials/arkode.h>
+
+#include "../tests.h"
+
+// Test repeated calls to solve_ode with harmonic oscillator problem
+
+using VectorType = Vector<double>;
+
+double kappa = 1.0;
+
+std::unique_ptr<SUNDIALS::ARKode<VectorType>>
+create_solver()
+{
+  ParameterHandler                             prm;
+  SUNDIALS::ARKode<VectorType>::AdditionalData data;
+  data.add_parameters(prm);
+
+  // Set to true to reset input file.
+  if (false)
+    {
+      std::ofstream ofile(SOURCE_DIR "/arkode_repeated_solve.prm");
+      prm.print_parameters(ofile, ParameterHandler::ShortText);
+      ofile.close();
+    }
+
+  std::ifstream ifile(SOURCE_DIR "/arkode_repeated_solve.prm");
+  prm.parse_input(ifile);
+
+  auto ode = std::make_unique<SUNDIALS::ARKode<VectorType>>(data);
+
+  ode->explicit_function =
+    [&](double, const VectorType &y, VectorType &ydot) -> int {
+    ydot[0] = y[1];
+    ydot[1] = -kappa * kappa * y[0];
```

Comment:
that just gives you a sine function, right?

Reply:
Correct, added a comment as well.